### PR TITLE
Add documentation and examples for the integration with Micrometer Tracing

### PR DESCRIPTION
**Pull request by violetagg**

### CODE REVIEW

**Comment by violetagg** [comment]
@reactor/netty-team @marcingrzejszczak @Buzzardo PTAL

**Review by marcingrzejszczak — APPROVED** [review]
Looks fantastic!

**Review by pderop — APPROVED** [review]
looks good to me

File: docs/asciidoc/http-client.adoc [hunk 1]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+
+The following table provides information for the HTTP client spans:
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| HTTP <HTTP METHOD> | Information and total time for the request/response.
+See <<observability-spans-http-client-response-span>>
+| hostname resolution | Information and time spent for resolving the address.
+See <<observability-spans-hostname-resolution-span>>
```
Comment (Buzzardo) [inline docs/asciidoc/http-client.adoc:449]: Add a period at the end.

File: docs/asciidoc/http-client.adoc [hunk 2]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+| connect | Information and time spent for connecting to the remote address.
+See <<observability-spans-connect-span>>
+| tls handshake | Information and time spent for TLS handshake.
+See <<observability-spans-tls-handshake-span>>
+|=======
+
+The following example enables that integration. This concrete example uses `Brave` and reports the information to `Zipkin`.
+See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup if needed.
```
Comment (Buzzardo) [inline docs/asciidoc/http-client.adoc:457]: Remove `if needed`. (That's always implied with a See or See also link.)

File: docs/asciidoc/http-client.adoc [hunk 3]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+
+====
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..90]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
```
Comment (Buzzardo) [inline docs/asciidoc/http-client.adoc:465]: Change `initialises` to `initializes`. (Spring uses American spelling.)

Add a comma after `Zipkin`. (Spring uses serial commas - often called the Harvard comma or the Oxford comma.)

File: docs/asciidoc/tcp-client.adoc [hunk 4]
```diff
@@ -321,6 +321,39 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..37]
...
+| hostname resolution | Information and time spent for resolving the address.
+See <<observability-spans-hostname-resolution-span>>
+| connect | Information and time spent for connecting to the remote address.
+See <<observability-spans-connect-span>>
+| tls handshake | Information and time spent for TLS handshake.
+See <<observability-spans-tls-handshake-span>>
+|=======
+
+The following example enables that integration. This concrete example uses `Brave` and reports the information to `Zipkin`.
+See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup if needed.
```
Comment (Buzzardo) [inline docs/asciidoc/tcp-client.adoc:341]: Remove if needed. (That's always implied with a See or See also link.)

File: docs/asciidoc/tcp-client.adoc [hunk 5]
```diff
@@ -321,6 +321,39 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..37]
...
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..81]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
+<2> Enables the built-in integration with Micrometer
+====
+
+The result in `Zipkin`:
```
Comment (Buzzardo) [inline docs/asciidoc/tcp-client.adoc:353]: Add `looks like` before the colon at the end. Spring avoids sentence fragments, so we need a verb (and the verb requires an object this time).

File: docs/asciidoc/tcp-server.adoc [hunk 6]
```diff
@@ -293,6 +293,35 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| tls handshake | Information and time spent for TLS handshake.
+See <<observability-spans-tls-handshake-span>>
+|=======
+
+The following example enables that integration. This concrete example uses `Brave` and reports the information to `Zipkin`.
+See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup if needed.
```
Comment (Buzzardo) [inline docs/asciidoc/tcp-server.adoc:309]: Remove if needed. (That's always implied with a See or See also link.)

File: docs/asciidoc/tcp-server.adoc [hunk 7]
```diff
@@ -293,6 +293,35 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..81]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
+<2> Enables the built-in integration with Micrometer
+====
+
+The result in `Zipkin`:
```
Comment (Buzzardo) [inline docs/asciidoc/tcp-server.adoc:321]: Add `looks like` before the colon at the end. Spring avoids sentence fragments, so we need a verb (and the verb requires an object this time).

File: docs/asciidoc/tcp-server.adoc [hunk 8]
```diff
@@ -293,6 +293,35 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+The following example enables that integration. This concrete example uses `Brave` and reports the information to `Zipkin`.
+See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup if needed.
+
+====
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..81]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
```
Comment (Buzzardo) [inline docs/asciidoc/tcp-server.adoc:317]: Change initialises to initializes. (Spring uses American spelling.)

Add a comma after Zipkin. (Spring uses serial commas - often called the Harvard comma or the Oxford comma.)

File: docs/asciidoc/tcp-server.adoc [hunk 9]
```diff
@@ -293,6 +293,35 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+<2> Enables the built-in integration with Micrometer
```
Comment (Buzzardo) [inline docs/asciidoc/tcp-server.adoc:318]: Add a period at the end.

File: docs/asciidoc/http-server.adoc [hunk 10]
```diff
@@ -617,6 +617,36 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..41]
...
+
+====
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..91]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
+<2> Templated URIs will be used as an URI tag value when possible
+<3> Enables the built-in integration with Micrometer
```
Comment (Buzzardo) [inline docs/asciidoc/http-server.adoc:643]: Add a period at the end.

File: reactor-netty-http-brave/README.md [hunk 11]
```diff
@@ -1,8 +1,7 @@
 # Brave instrumentation for Reactor Netty HTTP
 
 :exclamation: This module is **DEPRECATED** as of `Reactor Netty 1.1.0` and will be removed in `Reactor Netty 2.0.0`.
-Prefer using the standard `HttpClient/HttpServer` metrics functionality which has integration with
-[Micrometer Observation API](https://github.com/micrometer-metrics/micrometer/wiki/Migrating-to-new-1.10.0-Observation-API)
+Prefer using the standard `HttpClient/HttpServer` integration with [Micrometer Tracing](https://micrometer.io/docs/tracing).
```
Comment (Buzzardo) [inline reactor-netty-http-brave/README.md:4]: Change `Prefer using` to `We prefer to use`.

**Review by Buzzardo — CHANGES_REQUESTED** [review]

File: docs/asciidoc/http-client.adoc [hunk 12]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+== Tracing
+The HTTP client supports built-in integration with https://micrometer.io/docs/tracing[`Micrometer Tracing`].
+
+The following table provides information for the HTTP client spans:
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| HTTP <HTTP METHOD> | Information and total time for the request/response.
+See <<observability-spans-http-client-response-span>>
```
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:447]:
```suggestion
See <<observability-spans-http-client-response-span>>.
```

File: docs/asciidoc/http-client.adoc (hunk 1 above)
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:449]:
```suggestion
See <<observability-spans-hostname-resolution-span>>.
```

File: docs/asciidoc/http-client.adoc [hunk 13]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+| hostname resolution | Information and time spent for resolving the address.
+See <<observability-spans-hostname-resolution-span>>
+| connect | Information and time spent for connecting to the remote address.
+See <<observability-spans-connect-span>>
```
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:451]:
```suggestion
See <<observability-spans-connect-span>>.
```

File: docs/asciidoc/http-client.adoc [hunk 14]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+| tls handshake | Information and time spent for TLS handshake.
+See <<observability-spans-tls-handshake-span>>
```
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:453]:
```suggestion
See <<observability-spans-tls-handshake-span>>.
```

File: docs/asciidoc/http-client.adoc (hunk 2 above)
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:457]:
```suggestion
See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup.
```

File: docs/asciidoc/http-client.adoc (hunk 3 above)
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:465]:
```suggestion
<1> Initializes Brave, Zipkin, and the Observation registry.
```

File: docs/asciidoc/http-client.adoc [hunk 15]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup if needed.
+
+====
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..90]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
+<2> Templated URIs will be used as an URI tag value when possible
```
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:466]:
```suggestion
<2> Templated URIs are used as an URI tag value when possible.
```

File: docs/asciidoc/http-client.adoc [hunk 16]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+<3> Enables the built-in integration with Micrometer
```
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:467]:
```suggestion
<3> Enables the built-in integration with Micrometer.
```

File: docs/asciidoc/http-client.adoc [hunk 17]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+====
+
+The result in `Zipkin`:
```
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:470]:
```suggestion
The result in `Zipkin` looks like:
```

File: docs/asciidoc/http-server.adoc [hunk 18]
```diff
@@ -617,6 +617,36 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..41]
...
+== Tracing
+The HTTP server supports built-in integration with https://micrometer.io/docs/tracing[`Micrometer Tracing`].
+
+The following table provides information for the HTTP server spans:
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| <HTTP METHOD>_<URI> | Information and total time for the request/response.
+See <<observability-spans-http-server-response-span>>
```
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:629]:
```suggestion
See <<observability-spans-http-server-response-span>>.
```

File: docs/asciidoc/http-server.adoc [hunk 19]
```diff
@@ -617,6 +617,36 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..41]
...
+|=======
+
+The following example enables that integration. This concrete example uses `Brave` and reports the information to `Zipkin`.
+See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup if needed.
```
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:633]:
```suggestion
See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup.
```

File: docs/asciidoc/http-server.adoc [hunk 20]
```diff
@@ -617,6 +617,36 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..41]
...
+
+====
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..91]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
```
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:641]:
```suggestion
<1> Initializes Brave, Zipkin, and the Observation registry.
```

File: docs/asciidoc/http-server.adoc [hunk 21]
```diff
@@ -617,6 +617,36 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..41]
...
+<2> Templated URIs will be used as an URI tag value when possible
```
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:642]:
```suggestion
<2> Templated URIs are used as an URI tag value when possible.
```

File: docs/asciidoc/http-server.adoc (hunk 10 above)
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:643]:
```suggestion
<3> Enables the built-in integration with Micrometer.
```

File: docs/asciidoc/http-server.adoc [hunk 22]
```diff
@@ -617,6 +617,36 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..41]
...
+<3> Enables the built-in integration with Micrometer
+====
+
+The result in `Zipkin`:
```
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:646]:
```suggestion
The result in `Zipkin` looks like:
```

File: docs/asciidoc/tcp-client.adoc [hunk 23]
```diff
@@ -321,6 +321,39 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..37]
...
+== Tracing
+The TCP client supports built-in integration with https://micrometer.io/docs/tracing[`Micrometer Tracing`].
+
+The following table provides information for the TCP client spans:
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| hostname resolution | Information and time spent for resolving the address.
+See <<observability-spans-hostname-resolution-span>>
```
Comment (violetagg) [inline docs/asciidoc/tcp-client.adoc:333]:
```suggestion
See <<observability-spans-hostname-resolution-span>>.
```

File: docs/asciidoc/tcp-client.adoc [hunk 24]
```diff
@@ -321,6 +321,39 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..37]
...
+| connect | Information and time spent for connecting to the remote address.
+See <<observability-spans-connect-span>>
```
Comment (violetagg) [inline docs/asciidoc/tcp-client.adoc:335]:
```suggestion
See <<observability-spans-connect-span>>.
```

File: docs/asciidoc/tcp-client.adoc [hunk 25]
```diff
@@ -321,6 +321,39 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..37]
...
+| tls handshake | Information and time spent for TLS handshake.
+See <<observability-spans-tls-handshake-span>>
```
Comment (violetagg) [inline docs/asciidoc/tcp-client.adoc:337]:
```suggestion
See <<observability-spans-tls-handshake-span>>.
```

File: docs/asciidoc/tcp-client.adoc (hunk 4 above)
Comment (violetagg) [inline docs/asciidoc/tcp-client.adoc:341]:
```suggestion
See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup.
```

File: docs/asciidoc/tcp-client.adoc [hunk 26]
```diff
@@ -321,6 +321,39 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..37]
...
+The following example enables that integration. This concrete example uses `Brave` and reports the information to `Zipkin`.
+See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup if needed.
+
+====
+[source,java,indent=0]
+.{examplesdir}/tracing/Application.java
+----
+include::{examplesdir}/tracing/Application.java[lines=18..81]
+----
+<1> Initialises Brave, Zipkin and the Observation registry.
```
Comment (violetagg) [inline docs/asciidoc/tcp-client.adoc:349]:
```suggestion
<1> Initializes Brave, Zipkin, and the Observation registry.
```

File: docs/asciidoc/tcp-client.adoc [hunk 27]
```diff
@@ -321,6 +321,39 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..37]
...
+<2> Enables the built-in integration with Micrometer
```
Comment (violetagg) [inline docs/asciidoc/tcp-client.adoc:350]:
```suggestion
<2> Enables the built-in integration with Micrometer.
```

File: docs/asciidoc/tcp-client.adoc (hunk 5 above)
Comment (violetagg) [inline docs/asciidoc/tcp-client.adoc:353]:
```suggestion
The result in `Zipkin` looks like:
```

File: docs/asciidoc/tcp-server.adoc [hunk 28]
```diff
@@ -293,6 +293,35 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
+== Tracing
+The TCP server supports built-in integration with https://micrometer.io/docs/tracing[`Micrometer Tracing`].
+
+The following table provides information for the TCP server spans:
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| tls handshake | Information and time spent for TLS handshake.
+See <<observability-spans-tls-handshake-span>>
```
Comment (violetagg) [inline docs/asciidoc/tcp-server.adoc:305]:
```suggestion
See <<observability-spans-tls-handshake-span>>.
```

File: docs/asciidoc/tcp-server.adoc (hunk 6 above)
Comment (violetagg) [inline docs/asciidoc/tcp-server.adoc:309]:
```suggestion
See the https://micrometer.io/docs/tracing#_micrometer_tracing_opentelemetry_setup[`Micrometer Tracing`] documentation for `OpenTelemetry` setup.
```

File: docs/asciidoc/tcp-server.adoc (hunk 8 above)
Comment (violetagg) [inline docs/asciidoc/tcp-server.adoc:317]:
```suggestion
<1> Initializes Brave, Zipkin, and the Observation registry.
```

File: docs/asciidoc/tcp-server.adoc (hunk 9 above)
Comment (violetagg) [inline docs/asciidoc/tcp-server.adoc:318]:
```suggestion
<2> Enables the built-in integration with Micrometer.
```

File: docs/asciidoc/tcp-server.adoc (hunk 7 above)
Comment (violetagg) [inline docs/asciidoc/tcp-server.adoc:321]:
```suggestion
The result in `Zipkin` looks like:
```

File: reactor-netty-http-brave/README.md (hunk 11 above)
Comment (violetagg) [inline reactor-netty-http-brave/README.md:4]:
```suggestion
We prefer to use the standard `HttpClient/HttpServer` integration with [Micrometer Tracing](https://micrometer.io/docs/tracing).
```

File: docs/asciidoc/http-client.adoc [hunk 29]
```diff
@@ -435,6 +435,42 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..35]
...
 
+== Tracing
+The HTTP client supports built-in integration with https://micrometer.io/docs/tracing[`Micrometer Tracing`].
+
+The following table provides information for the HTTP client spans:
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| HTTP <HTTP METHOD> | Information and total time for the request/response.
```
Comment (violetagg) [inline docs/asciidoc/http-client.adoc:446]:
```suggestion
| HTTP <HTTP METHOD> | Information and total time for the request.
```

File: docs/asciidoc/http-server.adoc [hunk 30]
```diff
@@ -617,6 +617,36 @@ include::{examplesdir}/metrics/custom/Application.java[lines=18..41]
...
 
+== Tracing
+The HTTP server supports built-in integration with https://micrometer.io/docs/tracing[`Micrometer Tracing`].
+
+The following table provides information for the HTTP server spans:
+
+[width="100%",options="header"]
+|=======
+| contextual name | description
+| <HTTP METHOD>_<URI> | Information and total time for the request/response.
```
Comment (violetagg) [inline docs/asciidoc/http-server.adoc:628]:
```suggestion
| <HTTP METHOD>_<URI> | Information and total time for the request.
```

**Comment by violetagg** [comment]
@marcingrzejszczak @pderop @Buzzardo Thanks for the review!

**Pull request merged**